lib/transport.js

```diff
@@ -14,7 +14,20 @@ class Transport {
 
   connect(cb) {
     return this._openSocket().then(() => this._authenticate()).then(() => {
-      this.socket.on('data', cb)
+      var buffers = []
+      var rem, len
+      this.socket.on('data', (chunk) => {
```

Comment:
I'll keep length header at transport level. Proprietary wire protocol and Apollo protocol will be handled on higher level.